### PR TITLE
Add prefix for external service accounts

### PR DESCRIPTION
**Pull request by linoman**
**What is this feature?**

[Add a brief description of what the feature or update does.]

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/services/extsvcauth/extsvcaccounts/service.go [hunk 1]
```diff
@@ -16,6 +16,10 @@ import (
 	sa "github.com/grafana/grafana/pkg/services/serviceaccounts"
 )
 
+const (
+	extsvcPrefix = "extsvc"
```
Comment (gamab) [inline pkg/services/extsvcauth/extsvcaccounts/service.go:20]: I'd suggest placing it here: https://github.com/grafana/grafana/blob/7d99da377fe8ce537b3c4c88dba50e4b820288ce/pkg/services/extsvcauth/extsvcaccounts/models.go#L11

**Review by gamab — COMMENTED** [review]
You should also retrieve it with this new name: https://github.com/grafana/grafana/blob/7d99da377fe8ce537b3c4c88dba50e4b820288ce/pkg/services/extsvcauth/extsvcaccounts/service.go#L103

**Review by gamab — COMMENTED** [review]
I'd add the `-` to the prefix, otherwise it looks slightly weird.

image

We also need to fix the tests, I think it's just the expected parameters in the mock's calls.

**Comment by linoman** [comment]
@gamab agreed! I'm working on the tests at the moment

File: pkg/services/extsvcauth/extsvcaccounts/service.go (hunk 1 above)
Comment (linoman) [reply pkg/services/extsvcauth/extsvcaccounts/service.go]: Changed

File: pkg/services/extsvcauth/extsvcaccounts/models.go [hunk 2]
```diff
@@ -6,7 +6,8 @@ import (
 )
 
 const (
-	kvStoreType = "extsvc-token"
+	extsvcPrefix = "extsvc"
```
[the remark this answers is not in the document]
Comment (linoman) [reply pkg/services/extsvcauth/extsvcaccounts/models.go]: Changed

**Review by gamab — APPROVED** [review]

**Pull request merged**